test/test_binary_ufuncs.py

```diff
@@ -163,6 +163,61 @@ def test_not_broadcastable(self, device, dtype, op):
             )
             raise AssertionError(msg)
 
+    @ops(binary_ufuncs)
+    def test_type_promotion(self, device, dtype, op):
+        shapes = [(), (1,)]
+        other_dtypes = set(op.supported_dtypes(device)) - {dtype}
+
+        inputs = [
+            (
+                make_tensor(lhs_shape, device=device, dtype=dtype, **op.lhs_make_tensor_kwargs),
+                make_tensor(lhs_shape, device=device, dtype=rhs_dtype, **op.rhs_make_tensor_kwargs),
+            )
+            for lhs_shape, rhs_shape, rhs_dtype in itertools.product(shapes, shapes, other_dtypes)
+        ]
```

Comment:
This is quite large set of tests (184 for CPU / CUDA and the 6 supported ops). Although the shape of the operators is as small as possible, this might still introduce some slowdown. Anyway to improve this without significantly reducing the coverage?